### PR TITLE
PR - Parsing

### PR DESCRIPTION
**Pull request by l-devigne**
Refactor parsing logic, improve wall validation and map filling

This Pull Request introduces several improvements and fixes to the map parsing logic in our cub3d project.

# Key changes:

Fix: no_walls_alone logic

Resolved an issue where using get_next_line and closing/reopening the file didn't reset it as expected.

Replaced it with a more reliable logic that checks if a line starts with '0' or '1', and ensures it's not just a line with \n.

Feature: fill_single_line() function

Added a new helper function to fill the grid line by line, improving readability and modularity of the code.

New wall validation logic

For every '0' character, the parser now checks the Left, Right, Up, and Down directions to ensure it's surrounded by walls.

Triggers an error if a '0' is isolated — improving map integrity checks.

Memory management improvements

Fixed memory leaks when handling dynamically counted lines before the map.

Cleaned up several functions for better allocation/deallocation flow.

Refactored t_map structure parsing

Improved parsing and validation of the t_map struct.

File converted to char **

The map file is now read and stored as a 2D array of characters, making it easier to parse and process.

# Commit Summary
<details> <summary>Click to expand</summary>

fix (no_walls_alone): Fixed GNL behavior and wall-start detection logic

feature (fill_grid | fill_single_line): Added fill_single_line() to simplify map filling

memory leak correction: Improved dynamic line handling and fixed memory issues

new concept for checking walls: Added 4-direction wall validation around each '0'

color conversion: Parsing improvements and color value handling

parsing the struct t_map: Refactored and validated map structure

convert file to char **: Fully transitioned map file to char double pointer

</details>
# How to test

Run the parser with:

Valid maps according to the mandatory part

Malformed maps (missing walls, isolated 0s etc...)

Edge cases (multiple empty lines, incomplete headers, etc.)

Use tools like valgrind or leaks to check for memory issues

Try different map file structures to validate robustness

### CODE REVIEW

File: src/file_check.c [hunk 1]
```diff
@@ -39,132 +106,140 @@ bool	process_line_check(const char *line)
...
-		return (0);
-	return (i + 1);
-}
-
 bool	is_line_full_wall(char *line)
 {
 	if (!line)
 		return (false);
 	if (!process_line_check(line))
-		return(free(line), false);
```
[the remark this answers is not in the document]
Comment (l-devigne) [reply src/file_check.c:60]: useless function for now, we could remove it
